Use a sentinel in `LatestValueCache` to denote if the cache is empty

shsms opened the pull request:
This is necessary because `None` is a valid value that could have been
sent through a channel, so the having a `None` value might not always
mean that the cache is empty.

Closes https://github.com/frequenz-floss/frequenz-sdk-python/issues/820

shsms committed:
Use a sentinel in `LatestValueCache` to denote if the cache is empty

This is necessary because `None` is a valid value that could have been
sent through a channel, so the having a `None` value might not mean
that the cache is empty.

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

shsms committed:
Update uses of the `LatestValueCache` to follow its new API

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

shsms commented:
This is an internal fix for something that was introduced in this release cycle, so no release notes necessary.

daniel-zullo-frequenz commented on src/frequenz/sdk/_internal/_channels.py:
```diff
@@ -37,14 +41,35 @@ def __init__(self, receiver: Receiver[T]) -> None:
+            ValueError: If no value has been received yet.
+        """
+        if isinstance(self._latest_value, _Sentinel):
+            raise ValueError("No value has been received yet.")
```
I personally found it weird that a property raises. Maybe it should no longer be a property?

shsms replied:
That's true,  I've changed it to the `get` method.  That seems to go well with the other new `has_value` method as well.

daniel-zullo-frequenz replied:
cool, thanks! It does not need an entry into the release-notes because it is internal, right?

shsms replied:
yup, it is not exposed outside at the moment.

daniel-zullo-frequenz reviewed (commented):
I've only have a comment to check for. LGTM otherwise

shsms committed:
Replace `LatestValueCache.latest_value` property with a `get` method

The previous commit has introduced the possibility for exceptions to
be raised from the implementation of `latest_value` property.

Properties should ideally not raise, so we replace it with the `get`
method.

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>
